glance/api/v2/images.py

```diff
@@ -92,22 +93,51 @@ def __init__(self, db_api=None, policy_enforcer=None, notifier=None,
     def create(self, req, image, extra_properties, tags):
         image_factory = self.gateway.get_image_factory(req.context)
         image_repo = self.gateway.get_repo(req.context)
+
         try:
             if 'owner' not in image:
                 image['owner'] = req.context.project_id
 
-            api_policy.ImageAPIPolicy(req.context, image,
-                                      self.policy).add_image()
-
+            api_policy.ImageAPIPolicy(req.context, image, self.policy).add_image()
             ks_quota.enforce_image_count_total(req.context, req.context.owner)
-            image = image_factory.new_image(extra_properties=extra_properties,
-                                            tags=tags, **image)
+
+            # Attach domain info if enabled
+            if getattr(CONF, 'show_domain_info', False):
+                domain_info = utils.fetch_domain_info(image['owner'], auth_token=req.context.auth_token)
+                if domain_info:
+                    extra_properties['domain_id'] = str(domain_info.get('domain_id', ''))
+                    extra_properties['domain_name'] = str(domain_info.get('domain_name', ''))
+                    tags_list = domain_info.get('domain_tags', [])
+                    if isinstance(tags_list, list):
+                        extra_properties['domain_tags'] = ','.join(tags_list)
+                    else:
+                        extra_properties['domain_tags'] = str(tags_list)
+                    LOG.debug("Attached domain info to image during create (no ID): %s", domain_info)
+            else:
+                tags_list = getattr(req.context, 'domain_tags', None)
+                if tags_list:
+                    if isinstance(tags_list, list):
+                        extra_properties['domain_tags'] = ','.join(tags_list)
+                    else:
+                        extra_properties['domain_tags'] = str(tags_list)
+                    LOG.debug("Attached project tags to image during create (no ID): %s", tags_list)
+
+            # Add domain-project scoped tag for iaas domains
+            user_domain = getattr(req.context, 'user_domain_name', None)
+            project_id = getattr(req.context, 'project_id', None)
+            if user_domain and user_domain.startswith('iaas') and project_id:
+                scoped_tag = f"{user_domain}-{project_id}"
+                tags = tags or []
+                if scoped_tag not in tags:
+                    tags.append(scoped_tag)
+                    LOG.debug("[IAAS_CREATE] Appended scoped tag '%s' to image", scoped_tag)
```

Comment:
If I would create an image for a iaas-domain project with my ccadmin/cloud_admin user, the image would get the `ccadmin-123124142` tag. Is that expected behaviour for you? I would have expected that the domain of the owner i.e. the project would be relevant.